packages/signals/src/signal.rs

```diff
@@ -104,44 +166,49 @@ pub(crate) struct SignalData<T> {
 /// use dioxus_signals::*;
 ///
 /// #[component]
-/// fn App(cx: Scope) -> Element {
-///     let mut count = use_signal(cx, || 0);
+/// fn App() -> Element {
+///     let mut count = use_signal(|| 0);
 ///
 ///     // Because signals have automatic dependency tracking, if you never read them in a component, that component will not be re-rended when the signal is updated.
 ///     // The app component will never be rerendered in this example.
-///     render! { Child { state: count } }
+///     rsx! { Child { state: count } }
 /// }
 ///
 /// #[component]
-/// fn Child(cx: Scope, state: Signal<u32>) -> Element {
+/// fn Child(state: Signal<u32>) -> Element {
 ///     let state = *state;
 ///
-///     use_future!(cx,  |()| async move {
+///     use_future( |()| async move {
 ///         // Because the signal is a Copy type, we can use it in an async block without cloning it.
 ///         *state.write() += 1;
 ///     });
 ///
-///     render! {
+///     rsx! {
 ///         button {
 ///             onclick: move |_| *state.write() += 1,
 ///             "{state}"
 ///         }
 ///     }
 /// }
 /// ```
-pub struct Signal<T: 'static> {
-    pub(crate) inner: CopyValue<SignalData<T>>,
+pub struct Signal<T: 'static, S: Storage<SignalData<T>> = UnsyncStorage> {
+    pub(crate) inner: CopyValue<SignalData<T>, S>,
 }
 
+/// A signal that can safely shared between threads.
+pub type SyncSignal<T> = Signal<T, SyncStorage>;
+
 #[cfg(feature = "serde")]
-impl<T: serde::Serialize + 'static> serde::Serialize for Signal<T> {
+impl<T: serde::Serialize + 'static, S: Storage<SignalData<T>>> serde::Serialize for Signal<T, S> {
```

Comment:
This will generate the following error.

```
error[E0403]: the name `S` is already used for a generic parameter in this item's generic parameters
   --> C:\Users\atty\.cargo\git\checkouts\dioxus-1e619ce595d3799d\87dc409\packages\signals\src\signal.rs:203:18
    |
202 | impl<T: serde::Serialize + 'static, S: Storage<SignalData<T>>> serde::Serialize for Signal<T, S> {
    |                                     - first use of `S`
203 |     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
    |                  ^ already used
```

Reply:
Fixed by https://github.com/DioxusLabs/dioxus/pull/1791/commits/55b37cc381f832ee842e3d0a5c177db9b2e19498